management/models.py

```diff
@@ -25,6 +32,11 @@ class Cohort(BaseModel):
     group = models.CharField(max_length=50)
     active = models.BooleanField(default=True)
 
+    def delete(self, *args, **kwargs):
+        request = kwargs.pop("request", None)
+        create_event(EventNames.CohortDeleted,request = request)
```

Comment:
Minor comment but I'd track the creation after the deletion actually occurs (for instance after the super is called)

If there was some issue calling the super and the delete didn't happen then the create_event log message shouldn't be called. In this case it would be. 

This would also allow us to (possibly) have a try/except and log a different message if the deletion couldn't be performed.

Reply:
Makes sense.
Would using the "redirect" code be the best path?

example from tracking/utils.py
if response.status_code != 200:
            if response.status_code == 302:

Reply:
More complete example

`if response.status_code != 200:
            if response.status_code == 302:
                create_event(EventNames.Redirected, request, user=user,
                             note='From: %s \tTo: %s' % (request.path, response['Location']))
            else:
                create_event(EventNames.PageError, request, user=user,
                             note='\n'.join([request.path, 'Response Code: %d' % response.status_code,]))
        else:
            create_event(EventNames.PageViewed, request, user=user)`

Reply:
Well that's what the view would throw. That's fine if you have the deletion tracking after the super.

If super throws an exception nothing would be logged. I don't think we need to log anything in the event log if there is an exception here.

I was thinking for here at most a try/catch on exceptions that django delete might throw which just looks like ProtectedError. 

It's fine to just move the create_event after the super and something else can handle the exception.

Reply:
Updated.
Going to create another issue for Exceptions.
Thanks!

Reply:
Yeah I think there's probably more things we could track, but just tracking this is a nice thing to have. Really if I was going to expand this issue we should track it, but probably also have had a "soft delete" style process. The reason this came about in the first place is that we deleted a cohort but we didn't even know which one, and we had to go to the database back-ups to find it. 

Soft delete would be a lot more effort though since the delete deletes across foreign keys. 

I feel like this is the easiest way to implement that, implementing a new Manager to handle the soft deletes and intercept access to them. Pretty cool! 

https://medium.com/@adriennedomingus/soft-deletion-in-django-e4882581c340